client/src/components/authPages/SignIn.js

```diff
@@ -0,0 +1,113 @@
+import React, { Component } from "react";
+import { withApollo } from "react-apollo";
+import { connect } from "react-redux";
+import { Link } from "react-router-dom";
+import { userLoginQuery } from "../../schema/queries";
+import Header from "./Header";
+
+class SignIn extends Component {
+	state = {
+		email: "",
+		password: "",
+		error: ""
+	};
+
+	validate = () => {
+		const { email, password } = this.state;
+		if (email === "" || password === "") {
+			this.setState({
+				error: "email or password couldn't be empty"
+			});
+			return false;
+		}
+		return true;
+	};
+	handleSubmit = async e => {
+		e.preventDefault();
+		const { email, password } = this.state;
+		if (this.validate()) {
+			try {
+				const response = await this.props.client.query({
+					query: userLoginQuery,
+					variables: { input: { email, password } }
+				});
+				console.log(response);
+				this.props.login();
+				this.props.history.push("/");
+			} catch (err) {
+				const errMsg = err.message.split(":")[1].trim();
+				this.setState({ error: errMsg, password: "" });
+			}
+		}
+	};
+
+	render() {
+		// console.log(this.props);
+		return (
+			<div className="container">
+				<Header />
+				<p>SignIn</p>
+				<form onSubmit={this.handleSubmit}>
+					{this.state.error !== "" && (
+						<div className="alert alert-danger" role="alert">
+							{this.state.error}
+						</div>
+					)}
+
+					<div className="form-group">
+						<label htmlFor="exampleInputEmail1">
+							Email address
+						</label>
+						<input
```

Comment:
From what I see you have used Redux and using classic input components. 
Apollo client provides their own query/mutation components.
Any reason why we went that path?

I'm not saying that using redux and Apollo client indirectly is the wrong choice, but 
I would like to know why we have used that. Redux going to introduce lots of boilerplate for this small application. It is good fun to learn it though :) 

See: https://www.apollographql.com/docs/react/essentials/queries.html

Reply:
I've used redux just because it holds the global state whether any user is logged in or not to protect private routes.